stdlib/public/TensorFlow/PythonConversion.swift

```diff
@@ -164,4 +164,21 @@ extension Tensor where Scalar : NumpyScalarCompatible {
   public func makeNumpyArray() -> PythonObject { return array.makeNumpyArray() }
 }
 
+extension TensorShape : PythonConvertible {
+  public var pythonObject: PythonObject {
+    return dimensions.pythonObject
+  }
+
+  public init?(_ pythonObject: PythonObject) {
+    let hasLen = Bool(Python.hasattr(pythonObject, "__len__"))
+    if(hasLen == true) {
+      guard let array = [Int32](pythonObject) else { return nil }
+      self.init(array)
+    } else {
+      guard let num = Int32(pythonObject) else { return nil }
```

Comment:
Matching scalar does seem right to me. Why is this necessary?

Reply:
@rxwei the reason is that ```[Int32](1)``` (and thus ```[Int32](1 as PythonObject)```) is not valid in Swift, but ```TensorShape(1)``` is valid.  Or do you mean that you would prefer ```[Int32]([pythonObject])``` here?